crates/supervisor/core/src/syncnode/node.rs

```diff
@@ -285,6 +274,86 @@ impl ManagedNode {
 
         Ok(())
     }
+
+    /// Calls a ManagedNodeApi method using an HTTP client with JWT authentication.
+    ///
+    /// # Arguments
+    /// * `f` - A closure that takes an HTTP client and returns a future with the RPC call
+    ///
+    /// # Returns
+    /// * `RpcResult<T>` - The result of the RPC call or an appropriate error
+    pub async fn call_rpc<T, F, Fut>(&self, f: F) -> RpcResult<T>
+    where
+        F: FnOnce(jsonrpsee::http_client::HttpClient) -> Fut,
+        Fut: std::future::Future<Output = Result<T, jsonrpsee::core::ClientError>>,
+    {
+        let client = self.create_authenticated_client().await?;
+        self.execute_rpc_call(client, f).await
+    }
+
+    /// Creates an authenticated HTTP client with JWT headers.
+    async fn create_authenticated_client(&self) -> RpcResult<jsonrpsee::http_client::HttpClient> {
+        let http_url = format!("http://{}", self.config.url);
+        debug!(target: "managed_node", url = %http_url, "Creating authenticated HTTP client");
+
+        let headers = self.create_auth_headers()?;
+
+        HttpClientBuilder::default().set_headers(headers).build(&http_url).map_err(|err| {
+            let err_msg = format!("Failed to create HTTP client: {}", err);
+            error!(target: "managed_node", ?err, %err_msg);
+            ErrorObject::from(ErrorCode::InternalError)
+        })
+    }
+
+    /// Creates authentication headers using JWT secret.
+    fn create_auth_headers(&self) -> RpcResult<HeaderMap> {
+        let jwt_secret = self.config.jwt_secret().ok_or_else(|| {
+            error!(target: "managed_node", "JWT secret not found or invalid");
+            ErrorObject::from(ErrorCode::ParseError)
+        })?;
+
+        let mut headers = HeaderMap::new();
+        let auth_header =
+            format!("Bearer {}", alloy_primitives::hex::encode(jwt_secret.as_bytes()));
+
+        headers.insert(
+            "Authorization",
+            HeaderValue::from_str(&auth_header).map_err(|err| {
+                let err_msg = format!("Invalid authorization header: {}", err);
+                error!(target: "managed_node", ?err, %err_msg);
+                ErrorObject::from(ErrorCode::ParseError)
+            })?,
+        );
+
+        debug!(target: "managed_node", "Authentication headers created");
+        Ok(headers)
+    }
+
+    /// Executes the RPC call with error handling and logging.
+    ///
+    /// # Arguments
+    /// * `client` - The authenticated HTTP client
+    /// * `f` - The RPC call closure
+    async fn execute_rpc_call<T, F, Fut>(
+        &self,
+        client: jsonrpsee::http_client::HttpClient,
+        f: F,
+    ) -> RpcResult<T>
+    where
+        F: FnOnce(jsonrpsee::http_client::HttpClient) -> Fut,
+        Fut: std::future::Future<Output = Result<T, jsonrpsee::core::ClientError>>,
```

Comment:
```suggestion
        F: FnOnce(HttpClient) -> Fut,
        Fut: Future<Output = Result<T, jsonrpsee::core::ClientError>>,
```
no need to specify full path here, import in import prelude of module